### PR TITLE
[now-cli] Add `NOW_PROJECT_ID` and `NOW_ORG_ID` env variable

### PR DESCRIPTION
**Pull request by lucleray**
This PR adds `NOW_PROJECT_ID` and `NOW_ORG_ID` environment variables as a way to specify a project to deploy to when running `now`.

eg.
```
NOW_ORG_ID=abc NOW_PROJECT_ID=123 now
```

When using these environment variables:
- the folder is not linked to the project (ie. no `.now` is created)
- we interrupt the process and show an error message if the project is not found:
image

### CODE REVIEW

File: packages/now-cli/test/integration.js [hunk 1]
```diff
@@ -2237,6 +2237,40 @@ test('ensure `github` and `scope` are not sent to the API', async t => {
   t.is(output.exitCode, 0, formatOutput(output));
 });
 
+test.only('deploy with unknown `NOW_ORG_ID` and `NOW_PROJECT_ID`', async t => {
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]:
```suggestion
test('deploy with unknown `NOW_ORG_ID` and `NOW_PROJECT_ID`', async t => {
```

File: packages/now-cli/test/integration.js [hunk 2]
```diff
@@ -2237,6 +2237,40 @@ test('ensure `github` and `scope` are not sent to the API', async t => {
...
+      NOW_ORG_ID: 'asdf',
+      NOW_PROJECT_ID: 'asdf',
+    },
+  });
+
+  t.is(output.exitCode, 1, formatOutput(output));
+  t.is(output.stderr.includes('Project not found'), true, formatOutput(output));
+});
+
+test.only('deploy with `NOW_ORG_ID` and `NOW_PROJECT_ID`', async t => {
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]:
```suggestion
test('deploy with `NOW_ORG_ID` and `NOW_PROJECT_ID`', async t => {
```

**Review by TooTallNate — CHANGES_REQUESTED** [review]

File: packages/now-cli/test/integration.js (hunk 1 above)
Comment (lucleray) [reply packages/now-cli/test/integration.js]: I updated around the same time you commented 😅

File: packages/now-cli/src/util/projects/link.ts [hunk 3]
```diff
@@ -43,11 +36,21 @@ export async function getLinkedProject(
   path: string
 ): Promise<[Org | null, Project | null]> {
   try {
-    const json = await readFile(join(path, NOW_FOLDER, NOW_PROJECT_LINK_FILE), {
-      encoding: 'utf8',
-    });
+    let link: ProjectLink;
+
+    if (process.env.NOW_ORG_ID && process.env.NOW_PROJECT_ID) {
```
Comment (TooTallNate) [inline packages/now-cli/src/util/projects/link.ts]: Perhaps destructure these env vars first since they're accessed immediately within the `if` block?

Comment (lucleray) [reply packages/now-cli/src/util/projects/link.ts]: Sure 👍

**Review by TooTallNate — APPROVED** [review]

**Comment by honzajavorek** [comment]
Is this a documented and supported way how to deploy with Now CLI v17 non-interactively?

Upgrading to Now CLI v17 broke most of my deployments done by the CLI from my CI jobs. This is the missing piece which allows me to still run the Now CLI non-interactively, but as far as I'm looking, it's not mentioned anywhere in the user-facing docs.

------

**Update:** If anyone finds this comment, you need to set the following env vars and then you can run the Now CLI:

```
$ export NOW_ORG_ID=...
$ export NOW_PROJECT_ID=...
$ now
```

The trick is it doesn't work with values like `honzajavorek` and `honzajavorek.cz`, they really need to be IDs. There are no docs on how to get the IDs and they're not visible anywhere in the Dashboard. So you need to run `now` locally interactively, connect your project, and then inspect the locally created `.now` folder to get the values. I'm unsure whether the values should be treated as secrets.

**Comment by robertcoopercode** [comment]
Thanks @honzajavorek that cleared things up for me.

To anyone that want more details on how to deploy to multiple environments using Now CLI v17, I wrote up a post about it here: https://www.robertcooper.me/deploy-app-to-multiple-environments-zeit-now

**Comment by richardvanbergen** [comment]
Any chance we can update the documentation surrounding this? I've just wasted two hours trying to find a non-deprecated solution to specifying a project name. I get that this is hot-off-the-press but it should really be a note in the `Project Linking` section.

**Pull request merged**